File: backend-springboot/src/main/java/com/eventzen/controllers/AdminController.java
package com.eventzen.controllers;

import com.eventzen.entities.Booking;
import com.eventzen.entities.Budget;
import com.eventzen.repositories.BookingRepository;
import com.eventzen.repositories.BudgetRepository;
import com.eventzen.repositories.EventRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;

import java.util.HashMap;
import java.util.List;
import java.util.Map;

@RestController
@RequestMapping("/api/admin")
public class AdminController {

    @Autowired
    private BookingRepository bookingRepository;

    @Autowired
    private BudgetRepository budgetRepository;

    @Autowired
    private EventRepository eventRepository;

    // Dashboard Statistics
    @GetMapping("/dashboard")
    public Map<String, Object> getDashboardStats() {
        Map<String, Object> stats = new HashMap<>();
        stats.put("totalBookings", bookingRepository.count());
        stats.put("totalRevenue", budgetRepository.sumRevenue());
        stats.put("totalExpenses", budgetRepository.sumExpenses());
        stats.put("totalEvents", eventRepository.count());
        return stats;
    }

    // Fetch all bookings
    @GetMapping("/bookings")
    public List<Booking> getAllBookings() {
        return bookingRepository.findAll();
    }
}

File: backend-express/src/controllers/adminController.js
const Booking = require("../models/Booking");
const Budget = require("../models/Budget");
const Event = require("../models/Event");

// Get Dashboard Statistics
const getDashboardStats = async (req, res) => {
  try {
    const totalBookings = await Booking.count();
    const totalRevenue = await Booking.sum("total_price");
    const totalExpenses = await Budget.sum("actual_expense");
    const totalEvents = await Event.count();

    res.status(200).json({ totalBookings, totalRevenue, totalExpenses, totalEvents });
  } catch (error) {
    console.error("Dashboard stats error:", error);
    res.status(500).json({ message: "Error fetching dashboard stats", error });
  }
};

// Get All Bookings with Event Info
const getAllBookings = async (req, res) => {
  try {
    const bookings = await Booking.findAll({
      include: [{ model: Event, attributes: ["name"] }],
      order: [["createdAt", "DESC"]],
      limit: 5
    });

    res.status(200).json(bookings);
  } catch (error) {
    console.error("Bookings fetch error:", error);
    res.status(500).json({ message: "Error fetching bookings", error });
  }
};

module.exports = { getDashboardStats, getAllBookings };
